src/commands/github.ts

```diff
@@ -0,0 +1,270 @@
+import { CommandInteraction, GuildMember, MessageEmbed } from "discord.js";
+import fetch from "node-fetch";
+import type { Command } from "../command";
+import type { Config } from "../config";
+import { getOauth } from "../store/githubOauth";
+import type { Branch, Commit, GithubUser, Issue, PullRequest, Repository } from "../structures/github";
+
+const command: Command = {
+	name: "github",
+	description: "Helper command for retrieving data for the guild specified repository.",
+	options: [
+		{
+			type: "STRING",
+			name: "name",
+			description: "The type of item to fetch (branch, commit, issue, pullrequest)",
+			required: true,
+		},
+		{
+			type: "STRING",
+			name: "query",
+			description: "The query to fetch. (REF for commits, pull-request number, etc)",
+			required: true,
+		},
+		{
+			type: "STRING",
+			name: "repo",
+			description: "A custom repository to fetch data from",
+		},
+	],
+	hasPermission: (state, interaction) =>
+		(interaction.member as GuildMember).roles.cache.has(state.config.developerRoleId),
```

Comment:
Any reason why permission is restricted to just developers?

Reply:
Was for testing purposes. :kek:

Should be general use, yeah?

Reply:
Yeah general use should be okay for this

Reply:
Updated to general use in botchannel and dev use everywhere.